plugins/language_kotlin.lua

```diff
@@ -0,0 +1,125 @@
+-- mod-version:3
+
+-- https://kotlinlang.org/docs/keyword-reference.html
+
+local syntax = require "core.syntax"
+
+syntax.add {
+  name = "Kotlin",
+  files = { "%.kt$" },
+  comment = "//",
+  block_comment = { "/*", "*/" },
+  patterns = {
+    { pattern = "//.*",                              type = "comment"  }, -- Comment, single-line
+    { pattern = { "/%*", "%*/" },                    type = "comment"  }, -- Comment, multi-line
+    { pattern = { '"', '"', '\\' },                  type = "string"   }, -- String, quotation marks
+    { pattern = { "'", "'", '\\' },                  type = "string"   }, -- String, apices
+    { pattern = "'\\x%x?%x?%x?%x'",                  type = "string"   }, -- Character hexadecimal escape sequence
+    { pattern = "'\\u%x%x%x%x'",                     type = "string"   }, -- Character unicode escape sequence
+    { pattern = "'\\?.'",                            type = "string"   }, -- Character literal
+    { pattern = "-?0x%x+",                           type = "number"   }, -- ?
+    { pattern = "-?%d+[%deE]*f?",                    type = "number"   }, -- ?
+    { pattern = "-?%.?%d+f?",                        type = "number"   }, -- ?
+    { regex   = [[\-\>(?=\s)]],                      type = "operator" }, -- Lambda
+    { regex   = [[\.{2}\<?\s?(?=[\\-]?[a-z0-9])]],   type = "operator" }, -- Range operators
+    { pattern = "[%+%-=/%*%^%%<>!~|&]",              type = "operator" }, -- Operators
+    { regex   = [[\?(?=\.)]],                        type = "operator" }, -- ?. operator
+    { pattern = "[%a_][%w_]*%f[(]",                  type = "function" }, -- Function/Method/Class
+    { regex   = [[let(?=\s\{)]],                     type = "function" }, -- ? operator
+    { regex   = [[\?\:(?=\s?)]],                     type = "operator" }, -- elvis operator
+    { regex   = [[this(?=\.?\@?)]],                  type = "keyword"  }, -- this keyword
+    { regex   = [[\@[a-zA-Z]+]],                     type = "string"   }, -- Annotations
+    { regex   = [[[a-zA-Z]+\@(?=\s?[a-zA-Z])]],      type = "string"   }, -- Annotations (this pattern is lower priority than the `this keyword` pattern)
+    { regex   = "[A-Z][A-Z_]+",                      type = "keyword2" }, -- Constants, FULL UPPERCASE
+    { pattern = "[%a_][%w_]*",                       type = "symbol"   },
```

Comment:
```suggestion
    { pattern = "import()%s+()[%w_.]+",              type = { "keyword", "normal", "normal" } },
    { pattern = "[%a_][%w_]*",                       type = "symbol"   },
```
To avoid that parts of the imported "path" get highlighted:
![Schermata del 2024-04-15 19-26-51](https://github.com/lite-xl/lite-xl-plugins/assets/2798487/d402a453-2ef9-4920-aaed-2ee40d920f5c)